### PR TITLE
Tv show and Film pojos

### PR DESCRIPTION
**Pull request by evilduck95**
Adding a more structured approach to parsing and saving TV Shows

### CODE REVIEW

**Comment by craziii** [comment]
Went a little out of scope on branch, also covers Films

File: src/main/java/com/evilduck/filecatcher/service/FilmService.java [hunk 1]
```diff
@@ -1,22 +1,86 @@
...
+                if (testValue != resolution) {
+                    return Year.of(testValue);
+                }
+            }
+        }
+        return Year.of(Year.MIN_VALUE);
+    }
+
+    private String parseFilmName(final String filename){
+        // TODO: Add in parsing of film name, likely that the film name is going to be before the year or resolution
```
Comment (evilduck95) [inline src/main/java/com/evilduck/filecatcher/service/FilmService.java]: Wat do?

File: src/main/java/com/evilduck/filecatcher/service/TvShowService.java [hunk 2]
```diff
@@ -38,6 +129,21 @@ public void save(final Resource media, final String contentType) throws IOExcept
...
+//            if (isValidTvShowFolder(tempFolder)) {
+//                tvShowRepository.save(tempFolder, media.getFilename().replaceFirst("[.].+", ""));
+//            } else {
+//                throw new IncorrectFileFormatException("Unable to find at least one video file in every season");
+//            }
+//        } else {
+//            throw new IncorrectFileFormatException("File is not a ZIP archive");
+//        }
+//    }
+
```
Comment (evilduck95) [inline src/main/java/com/evilduck/filecatcher/service/TvShowService.java]: I must learn to cleanup after myself :3

**Review by evilduck95 — COMMENTED** [review]
All looks pog af

**Pull request merged**